runtime/compiler/optimizer/J9TransformUtil.cpp

```diff
@@ -40,6 +40,21 @@
 #include "j9.h"
 #include "optimizer/OMROptimization_inlines.hpp"
 
+/**
+ * Walks the TR_RegionStructure counting loops to get the nesting depth of the block
+ */ 
+int32_t J9::TransformUtil::getLoopNestingDepth(TR::Compilation *comp, TR::Block *block)
+   {
+   TR_RegionStructure *region = block->getParentStructureIfExists(comp->getFlowGraph());
+   int32_t nestingDepth = 0;
+   while (region && !region->isAcyclic())
```

Comment:
This means you count improper regions as loops for the purposes of your loop test - is this intentional?

Reply:
I walked over the parent structure till I found acyclic region. Checking the function it will return true when we have a region that does not contain any internal cycles and does not have `backedge`.

Reply:
Right but do you want to count an improper region - eg one that contains cycles, but which is not a natural loop as a loop for scaling - that seems wrong to me...

Reply:
Thanks for explanation. Changed it in https://github.com/eclipse/openj9/pull/2348/commits/113d8283b3fdeb03d4aa6a8ed62d1185301d2a08 
Now routine walks parent structure till it finds a regions that is not natural looop